R/France.R

```diff
@@ -0,0 +1,142 @@
+#' France Class containing country specific attributes and methods
+#'
+#' @description Country specific information for downloading, cleaning
+#' and processing covid-19 region data for France.
+#'
+#' Data is sourced from \url{https://health-infobase.France.ca}.
+#' @details Inherits from `DataClass`
+#' @examples
+#' \dontrun{
+#' region <- France$new(verbose = TRUE, steps = TRUE)
+#' region$download()
+#' region$clean()
+#' region$process()
+#' region$return()
+#' }
+France <- R6::R6Class("France",
+  inherit = DataClass,
+  public = list(
+
+    # Core Attributes (amend each paramater for country specific infomation)
+    #' @field level_1_region the level 1 region name.
+    level_1_region = "region",
+    #' @field level_2_region the level 1 region name.
+    level_2_region = "departement",
+    #' @field data_url link to raw data for cases
+    data_url = "https://www.data.gouv.fr/fr/datasets/r/406c6a23-e283-4300-9484-54e78c8ae675", # nolint
+    #' @field source_data_cols existing columns within the raw data
+    source_data_cols = c("cases_new", "deaths_new",
+                         "hosp_new", "tested_new"),
+    #' @field hosp_url link to raw data for hospitalisations
+    hosp_url = "https://www.data.gouv.fr/fr/datasets/r/6fadff46-9efd-4c53-942a-54aca783c30c", # nolint
+
+
+    #' @description France-specific function for downloading raw data.
+    download = function() {
+      if (self$level == 1) {
+        self$data_url <- "https://www.data.gouv.fr/fr/datasets/r/001aca18-df6a-45c8-89e6-f82d689e6c01" # nolint
+      }
+      if (self$verbose) {
```

Comment:
I'll add a utility for this to reduce the code repetition for those adding data.

Reply:
Great. I also went down a rabbit-hole at one point thinking that I'd want to have access to both levels of the `region_code`  tables. Which made me realize that `get_region_codes` does a lot more than just getting region codes and honestly has several other side effects.

Reply:
Yes that makes sense. I think it is sensible to want both levels of the codes (as level 2 data can have level 1 codes as well). We need a rethink of how this works before `dev` gets merged into `master`. There is an issue for this here: #171

Reply:
New approach to `region_codes` has been applied now.